### ISSUE TITLE
Explore a way to load .js files into 'window' thread type

### ISSUE THREAD

**Issue by wilsonpage** [gaia-components/threads#20]
Currently users have to define `.html` sources for 'window' type threads. This is not ideal as it means that scripts can't be moved between thread types easily.
#### Open questions:
1. If scripts have dependencies these have to be loaded differently depending on the env (eg. `importScripts` vs async script loading). **How can users author once and run anywhere?** Requirejs can run in both window and worker context, so this might be handy.
2. For window at least we'd have to have a 'container' page to load user's scripts into. What does this look like? Is it a real file or a memory based `Blob`?
3. Should single threads be able to load multiple sources, or should users' scripts always run in dedicated thread?

**Comment by wilsonpage** [comment]
``` js
var src = 'my-service.js';
var base = document.location.origin;
var html = '<script src="' + base + '/threads.js"><\/script><script src="' + base + '/' + src + '"><\/script>';
var blob = new Blob([html], {type: "text/html"});
var iframe = document.createElement('iframe');
iframe.src = URL.createObjectURL(blob);
document.body.appendChild(iframe);
```

**Comment by gmarty** [comment]
What about the following?

``` javascript
iframe.src = 'data:text/html;utf-8,<!DOCTYPE html><script defer src="' + base + '/threads.js"><script defer src="' + base + '/' + src + '"></script>';
```

Also, note the `defer` attribute.

**Comment by wilsonpage** [comment]
Nice! Even simpler! I had no idea you could write data-uris like that!

**Comment by wilsonpage** [comment]
After some discussion with @arcturus we've decided that the for reasons of limiting 'magic' `theads.js` should not be auto-loaded into the window. It seems like it would be a slippery slope to adding a lot of script loading cruft to the library.

[I wrote a gist](https://gist.github.com/wilsonpage/fcc5c6cf66506a9dfe2f) to illustrate how it's possible to run the same service in a `Worker` or a `Window`.
